backend/src/test/java/com/redhat/cloud/notifications/db/BehaviorGroupResourcesTest.java

```diff
@@ -45,245 +45,321 @@
+                                                        .onItem().call(behaviorGroup3 -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup1.getId())
+                                                                .chain(() -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup1.getId()))
+                                                                .chain(() -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup1.getId(), behaviorGroup2.getId()))
+                                                                .chain(() -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup2.getId()))
+                                                                .chain(() -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup1.getId(), behaviorGroup2.getId(), behaviorGroup3.getId()))
+                                                                .chain(() -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true))
+                                                        )
+                                                )
+                                        )
+                                )
+                        )
+                )
+                .await().indefinitely();
     }
 
     @Test
-    public void testFindEventTypesByBehaviorGroupId() {
-        Bundle bundle = resourceHelpers.createBundle();
-        Application app = resourceHelpers.createApplication(bundle.getId());
-        EventType eventType = createEventType(app.getId());
-        BehaviorGroup behaviorGroup = resourceHelpers.createBehaviorGroup(DEFAULT_ACCOUNT_ID, "displayName", bundle.getId());
-        updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup.getId());
-        List<EventType> eventTypes = resourceHelpers.findEventTypesByBehaviorGroupId(behaviorGroup.getId());
-        assertEquals(1, eventTypes.size());
-        assertEquals(eventType.getId(), eventTypes.get(0).getId());
+    void testFindEventTypesByBehaviorGroupId() {
+        resourceHelpers.createBundle()
+                .onItem().call(bundle -> resourceHelpers.createApplication(bundle.getId())
+                        .onItem().call(app -> createEventType(app.getId())
+                                .onItem().call(eventType -> resourceHelpers.createBehaviorGroup(DEFAULT_ACCOUNT_ID, "displayName", bundle.getId())
+                                        .onItem().call(behaviorGroup -> updateAndCheckEventTypeBehaviors(DEFAULT_ACCOUNT_ID, eventType.getId(), true, behaviorGroup.getId())
+                                                .chain(() -> resourceHelpers.findEventTypesByBehaviorGroupId(behaviorGroup.getId())
+                                                        .onItem().invoke(eventTypes -> {
```

Comment:
This starts to get really difficult to read (nesting too much). Is there any other way to write this up?

Reply:
I already wrote as many nested instructions as possible, these are all mandatory unfortunately. It could be broken into methods but I don't think that would really serve readability.

Reply:
I made a tiny improvement (shortcuts) but it doesn't affect the crazy nesting.

Can we merge this as is and maybe improve the readability later?